src/samples/audio/kwd_nn_detect_test.c

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright 2021 NXP
+//
+// Author: Cristina Feies <cristina.ilie@nxp.com>
+// Author: Viorel Suman <viorel.suman@nxp.com>
+
+#include <sof/samples/audio/kwd_nn_detect_test.h>
+
+static __aligned(64) uint8_t preprocessed_data[KWD_NN_CONFIG_PREPROCESSED_SIZE];
+static __aligned(8) uint8_t confidences[KWD_NN_CONFIDENCES_SIZE];
+
+static int kwd_nn_detect_postprocess(uint8_t confidences[KWD_NN_CONFIDENCES_SIZE])
+{
+	int i;
+	uint8_t max_confidence = confidences[0];
+	int max_idx = 0;
+
+	for (i = 1; i < KWD_NN_CONFIDENCES_SIZE; i++) {
+		if (max_confidence < confidences[i]) {
+			max_confidence = confidences[i];
+			max_idx = i;
+		}
+	}
+	if (max_confidence < KWD_NN_MAX_CONFIDENCE_THRESHOLD)
+		max_idx = 1;
+	return max_idx;
+}
+
+void kwd_nn_detect_test(struct comp_dev *dev,
+				const struct audio_stream *source,
+				uint32_t frames)
+{
+	void *src;
+	uint32_t count = frames; /**< Assuming single channel */
+	uint32_t sample;
+	uint32_t cycles_per_frame; /**< Clock cycles required per frame */
+	uint32_t one_sec_samples = KWD_NN_KEY_LEN;
+	uint32_t half_sec_samples = one_sec_samples / N;
+
+	/* synthetic load */
+	if (test_keyword_get_load_mips(dev)) {
+		/* assuming count is a processing frame size in samples */
+		cycles_per_frame = (test_keyword_get_load_mips(dev) * 1000000 * count)
+				/ source->rate;
```

Comment:
I don't know if we use 64-bit arithmetic here. 32-bit arithmetic will likely overflow in this. Just to note in case you use this and you get unexpected results.